integration/operator_test.go

```diff
@@ -0,0 +1,232 @@
+// +build k8srequired
+
+package integration
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"os/exec"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/cenk/backoff"
+	"github.com/giantswarm/certificatetpr"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	v1 "k8s.io/client-go/pkg/api/v1"
+)
+
+const (
+	certOperatorValuesFile = "/tmp/cert-operator-install.yaml"
+	defaultDeadline        = 15
+)
+
+var cs kubernetes.Interface
+
+// TestMain allows us to have common setup and teardown steps that are run
+// once for all the tests https://golang.org/pkg/testing/#hdr-Main.
+func TestMain(m *testing.M) {
+	var v int
+	var err error
+	cs, err = getK8sClient()
+	if err != nil {
+		v = 1
+		log.Printf("unexpected error: %v\n", err)
+	}
+
+	if err := setUp(cs); err != nil {
+		v = 1
+		log.Printf("unexpected error: %v\n", err)
+	}
+
+	if v == 0 {
+		v = m.Run()
```

Comment:
Where is `Run` coming from and what is it doing?

Reply:
It's a method of the `m` parameter passed to `TestMain`, there's a link in the comment https://golang.org/pkg/testing/#hdr-Main